### ISSUE TITLE
Want to try Margin Sample Mining Loss (MSML)

### ISSUE THREAD

**Issue by ExtraYin** [VisualComputingInstitute/triplet-reid#48]
Hi @lucasb-eyer  and @Pandoro , I want to try a different loss from arxiv.org/abs/1710.00478. According to my understanding, I only need to change the line 

> diff = furthest_positive - closest_negative

to

> diff = tf.reduce_max(furthest_positive) - tf.reduce_min(closest_negative)

Is this understanding correct? Thanks in advance!

**Comment by Pandoro** [comment]
Hello @ExtraYin 

quickly glancing over our code again this should indeed be the case.  But only the changes that are needed for the code. To get any useful results out of your experiments you should definitely **optimize the learning rate schedule** for the new loss!

However, in the paper the main benefit of this loss is seen when using an Xception network while using a single learning rate schedule that goes "until convergence". First of all to a rigorous comparison, every network and loss combination get it's own learning rate schedule. Especially when making such a change to the loss, the loss will likely result in bigger values and thus the learning rate that is optimal for the MSML loss will likely not be optimal for the batch hard loss and the other way around.

The other big "issue" I see with the paper is that they train a single model on all datasets together while still only managing to slightly outperform our initial batch hard results. And this given that they added more data augmentation. My gut feeling tells me these changes should have resulted in an mAP well over 74% on Market-1501. In total I am thus a little skeptical about the results of the paper and I wouldn't be surprised if just making your proposed change and leaving everything as is will not result in the performance gains described in the paper. But we have not tried this ourselves as far as I know.

**Comment by lucasb-eyer** [comment]
In case you go ahead with this, and do manage to get reasonably good results, we would be interested in hearing about it!
